### PR TITLE
Changed from google test to catch

### PR DESCRIPTION
**Pull request by teseoch**
Still missing to port `INSTANTIATE_TEST_CASE_P`

#### Check all that apply (change to `[x]`)
- [x] All changes meet [libigl style-guidelines](https://libigl.github.io/style-guidelines/).
- [x] Adds new .cpp file.
- [ ] Adds corresponding unit test.
- [ ] Adds corresponding python binding.
- [ ] This is a minor change.

### CODE REVIEW

File: tests/include/igl/avg_edge_length.cpp [hunk 1]
```diff
@@ -31,32 +31,32 @@ TEST(avg_edge_length, cube)
   double avg;
 
   avg = igl::avg_edge_length(V,F);
-  ASSERT_NEAR((12.*sqrt(side_sq) + 6.*sqrt(diag_sq))/(12.+6.), avg, epsilon);
+  REQUIRE (avg == Approx ((12.*sqrt(side_sq) + 6.*sqrt(diag_sq))/(12.+6.)).margin( epsilon));
```
[the remark this answers is not in the document]
Comment (teseoch) [reply tests/include/igl/avg_edge_length.cpp]: this part was autogenerated with a script. There might be a lot of cases like that. I dont think it is worth to time investment

**Comment by teseoch** [comment]
I fixed the couple of stuff in the cmake and reneabled the test with parameters.

With respect to the old version, it generates only 1 test case per set of parameters. I think it makes more sense.

File: tests/include/igl/avg_edge_length.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdumas) [reply tests/include/igl/avg_edge_length.cpp]: `find tests/ -exec sed -i 's/REQUIRE (/REQUIRE(/g' {} \;` (may need to escape the `(`).

[the remark this answers is not in the document]
Comment (jdumas) [reply tests/include/igl/avg_edge_length.cpp]: On mac you need `sed -i "" 's/...'` instead.

**Comment by danielepanozzo** [comment]
Mmm it seems to me that all tests have been ported (except bbw on windows which has some issues, but the test passes on mac and linux). @teseoch could you confirm? If that's the case I think we should merge this ASAP and polish it later. This PR is blocking all the other ones that require unit tests.

**Comment by alecjacobson** [comment]
We also need a well documented conversion script for PRs or other existing
gtests.

**Comment by jdumas** [comment]
There's only one other PRs (for embree). But the documentation on the website for the unit tests is a bit outdated (it still refers to the old repo).

**Comment by alecjacobson** [comment]
I have many tests that I haven’t pushed yet. We should document the
conversion process.

**Comment by teseoch** [comment]
I have used this script to convert gtest into catch2 testes
[MigrateFromGoogleTestToCatch.py.zip](https://github.com/libigl/libigl/files/2507701/MigrateFromGoogleTestToCatch.py.zip)

**Comment by danielepanozzo** [comment]
This looks good to me and the script is a good way to convert old tests. @alecjacobson can we merge? I would like to enable the testing on windows which depends on this PR.

**Comment by jdumas** [comment]
Enabling the testing on Windows is independent from this PR, it's just a matter of setting proper flags in the CMake.

**Comment by fwilliams** [comment]
Can we merge #952 (which uses Google Test) before we merge this PR?

**Comment by teseoch** [comment]
@alecjacobson can you review your review?

File: cmake/LibiglDownloadExternal.cmake [hunk 2]
```diff
@@ -134,10 +134,10 @@ function(igl_download_triangle)
 endfunction()
 
 ## Google test
```
Comment (qnzhou) [inline cmake/LibiglDownloadExternal.cmake]: Out of date comment.

**Comment by qnzhou** [comment]
I also have some unit tests that I want to create and push, but maybe it is better to wait until this pr is merged?

**Comment by jdumas** [comment]
It'd make sense to merge this one first and then convert existing PRs and use Catch2 for writing new tests.

**Comment by qnzhou** [comment]
Also, catch2 has the interesting [tag](https://github.com/libigl/libigl-legacy.git) feature.  Currently, we are using the same tag, `[igl]`, for most test cases.  We could create some convention on tagging unit tests so it is easier to run a specific subset of tests.

**Comment by jdumas** [comment]
Agreed. Can have one tag per module for now (core, cgal, embree, etc.).

**Comment by teseoch** [comment]
i am using the suite name for the different modules eg `TEST_CASE("CSGTree: extrusion", "[igl/copyleft/cgal]")`

**Comment by alecjacobson** [comment]
@teseoch does that script work for TEST and TEST_P ?

**Comment by teseoch** [comment]
@alecjacobson no the script dosent work with TEST_P.

To change the TEST_P you need to move the content of the TEST_P in a lambda like `const auto test_case = [](const std::string &param)`, then call `test_common::run_test_cases(test_common::all_meshes(), test_case);` where the first parameter is the stuff inside `INSTANTIATE_TEST_CASE_P`.

You can look at the this file:
https://github.com/libigl/libigl/pull/961/files#diff-10f6cbc342547d20dcb837dc856f9912
to see the small changes

**Comment by alecjacobson** [comment]
Is it possible to script this? There are some new tests that need to be converted.

**Comment by jdumas** [comment]
Unless you have >10 new tests that use `TEST_P`, it might be faster to convert them by hand. The script converts the normal `TEST` though.

**Comment by teseoch** [comment]
can we merge before pushing new tests?

**Pull request merged**